### PR TITLE
Setting-And-UI Update-->Pre Code Driven UI - PLEASE ACCEPT

### PR DESCRIPTION
**Pull request by dustinwloring1988**
This is the last time I will change the SCENE file before we start on a code driven UI. The settings file started using ...addListener(....) to help in the future.
Also add the trade test to the main SCENE by putting a button next to path finding

### CODE REVIEW

File: Assets/Scripts/Settings/GameSettings.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿public class GameSettings {
```
Comment (OzieGamma) [inline Assets/Scripts/Settings/GameSettings.cs]: brace on the next line

File: Assets/Scripts/UI/SettingHelper.cs [hunk 2]
```diff
@@ -12,25 +12,133 @@ public class SettingHelper : MonoBehaviour {
...
+	void OnEnable()
+	{
+		//create an instance of this class
+		gameSettings = new GameSettings ();
+
+		//get all avalible resolution for the display
+		myResolutions = Screen.resolutions;
+
+		//Add our listeners
+		fpsToggle.onValueChanged.AddListener(delegate { OnFPSToggle(); } );
```
Comment (OzieGamma) [inline Assets/Scripts/UI/SettingHelper.cs]: `fpsToggle.onValueChanged.AddListener(OnFPSToggle);` is shorter and should also work

File: Assets/Scripts/UI/SettingHelper.cs [hunk 3]
```diff
@@ -12,25 +12,133 @@ public class SettingHelper : MonoBehaviour {
...
+		langToggle.onValueChanged.AddListener(delegate { OnLangageToggle(); } );
+		fullScreenToggle.onValueChanged.AddListener(delegate { OnFullScreenToggle(); } );
+
+		resolutionDropdown.onValueChanged.AddListener(delegate { OnResolutionChange(); } );
+		aliasingDropdown.onValueChanged.AddListener(delegate { OnAliasingChange(); } );
+		vSyncDropdown.onValueChanged.AddListener(delegate { OnVSyncChange(); } );
+		quilityDropdown.onValueChanged.AddListener(delegate { OnQuilityChange(); } );
+
+		//masterVolume.onValueChanged.AddListener(delegate { OnMasterChange(); } );
```
Comment (OzieGamma) [inline Assets/Scripts/UI/SettingHelper.cs]: No commented code please.

File: Assets/Scripts/UI/SettingHelper.cs [hunk 4]
```diff
@@ -12,25 +12,133 @@ public class SettingHelper : MonoBehaviour {
...
+		fullScreenToggle.isOn = Screen.fullScreen;
+		fpsToggle.isOn = fpsObject.activeSelf;
+		langToggle.isOn =langDropDown.activeSelf;
+	}
+
+	void CreateResDropDown()
+	{
+		/*
+		for (int i = 0; i < myResolutions.Length; i++) {
+			resolutionDropdown.AddOptions (myResolutions [i].height.ToString () + "x" + myResolutions [i].width.ToString () + "   " + myResolutions [i].refreshRate.ToString ());
```
Comment (OzieGamma) [inline Assets/Scripts/UI/SettingHelper.cs]: No commented code please.

**Comment by OzieGamma** [comment]
Thanks for the PR!

What did you change in the scene? Are those changes needed?

**Comment by dustinwloring1988** [comment]
@OzieGamma Yes made changes to the setting panel and added a button to the options next to path finding for the trade test someone made. And removed commented code and the shorter way of adding a listener did not work.....

**Comment by OzieGamma** [comment]
@dustinwloring1988 Sorry about that, my bad.

**Comment by dustinwloring1988** [comment]
@OzieGamma No its fine Im not the best programmer and any help right or wrong will help me learn..

**Comment by dustinwloring1988** [comment]
Please add so we can start Code Driven UI

**Comment by OzieGamma** [comment]
I don't have Unity on my machine, so I can't try it out but I try to give help with my C# experience. That's how you learn !

**Comment by dustinwloring1988** [comment]
https://github.com/TeamPorcupine/ProjectPorcupine/issues/404  -  code driven UI coming soon......

**Comment by dustinwloring1988** [comment]
Added more code driven UI calls

**Comment by dustinwloring1988** [comment]
made a separate unity scene until it is complete

**Pull request closed without merging**